packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -122,3 +122,25 @@ class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkIm
   @override
   String toString() => '$runtimeType("$url", scale: $scale)';
 }
+
+/// Thrown to indicate that loading a network image failed
+///
+class NetworkImageLoadException implements Exception {
+  /// Creates a [NetworkImageLoadException] with the specified http status
+  /// [code] and the [uri]
+  NetworkImageLoadException({@required this.code, @required this.uri})
+    : assert(uri != null),
+      message = 'HTTP request failed, statusCode: $code, $uri';
+
+  /// An http status code.
```

Comment:
This should describe the relationship of this status code to the request, and also if it is nullable.